core/src/test/scala/bekindrewind/VcrClientSpec.scala

```diff
@@ -0,0 +1,63 @@
+package bekindrewind
+
+import munit._
+import io.circe.parser._
+import io.circe.syntax._
+
+import java.net.URI
+import java.nio.file._
+import java.time.OffsetDateTime
+import scala.jdk.CollectionConverters._
+
+class VcrClientSpec extends FunSuite {
+
+  test("Request and response are saved as JSON") {
+    val recordingPath = Files.createTempFile("test", ".json")
+    val client        = MockClient(recordingPath, RecordOptions.default, VcrMatcher(_ => true))
+    assert(client.previouslyRecorded.isEmpty)
+    assert(client.newlyRecorded.get().isEmpty)
+
+    val record = VcrRecord(
+      VcrRecordRequest("GET", new URI("https://example.com/foo.json"), "{}", Map.empty),
+      VcrRecordResponse(200, "ok", Map.empty, "{}"),
+      OffsetDateTime.parse("2100-05-06T12:34:56.789Z")
+    )
+    client.newlyRecorded.set(Vector(record))
+    client.save()
+    assert(client.previouslyRecorded.isEmpty)
+    assert(client.newlyRecorded.get().sizeIs == 1)
+
+    val savedJson = Files.readAllLines(recordingPath).asScala.mkString("")
+    val decoded   = decode[VcrRecords](savedJson).map(_.records)
+    assertEquals(decoded, Right(Vector(record)))
+  }
+
+  test("Client loads the previous record when being constructed") {
+    val req     = VcrRecordRequest("GET", new URI("https://example.com/foo.json"), "{}", Map.empty)
+    val res     = VcrRecordResponse(200, "ok", Map.empty, "{}")
+    val at      = OffsetDateTime.parse("2100-05-06T12:34:56.789Z")
+    val rawJson = s"""{ "version": "0.1.0",
+                     |  "records": [
+                     |    { "request": ${req.asJson},
+                     |      "response": ${res.asJson},
+                     |      "recordedAt": ${at.asJson}
+                     |    }
+                     |  ]
+                     |}""".stripMargin
```

Comment:
I'd probably build the JSON string like this:

```scala
val rawJson = VcrRecords(Vector(VcrRecord(req, res, at)), "0.1.0").asJson.spaces2
```

So that if the model changes it'll be a compile error.

But I see value in eventually adding tests with raw strings for when/if I update the serialization format, like adding/removing/changing fields (though hopefully I don't need to do breaking changes like that after this gets released).

Reply:
Fixed in https://github.com/reibitto/be-kind-rewind/pull/7/commits/7c81ae4174519ef067ded75945ee82cea6d6cf79